### PR TITLE
Support passing additional props to dashboard cards

### PR DESCRIPTION
**Pull request by rawagner**
@honza

### CODE REVIEW

File: frontend/public/components/dashboards-page/with-dashboard-resources.tsx [hunk 1]
```diff
@@ -31,71 +30,70 @@ const mapStateToProps = (state: RootState) => ({
...
+  stopWatchK8sResource: StopWatchK8sResource = resource => {
+    this.setState((state: WithDashboardResourcesState) => ({
+      k8sResources: state.k8sResources.filter(r => r.prop !== resource.prop),
+    }));
+  }
+
+  render() {
+    return (
+      <Firehose resources={this.state.k8sResources}>
+        {inject(this.props.children, {
```
Comment (spadgett) [inline frontend/public/components/dashboards-page/with-dashboard-resources.tsx]: It looks like we'll lose type checking with this change :(

Comment (rawagner) [reply frontend/public/components/dashboards-page/with-dashboard-resources.tsx]: Yep :( I wanted to do this to keep the type checking
```suggestion
   <WithDashboardResources>
    {(props: DashboardItemProps) => <DetailsCard_ {...props} />}
  </WithDashboardResources>
```

but that does not go well with inject's `React.Children.map` as that ignores function as children.

Im not sure how else to do this to keep types in `DashboardItemProps` as required

Comment (rawagner) [reply frontend/public/components/dashboards-page/with-dashboard-resources.tsx]: @spadgett  Latest change keeps types as before

File: frontend/public/components/dashboards-page/with-dashboard-resources.tsx [hunk 2]
```diff
@@ -31,8 +30,8 @@ const mapStateToProps = (state: RootState) => ({
   [RESULTS_TYPE.PROMETHEUS]: state.dashboards.get(RESULTS_TYPE.PROMETHEUS),
 });
 
-const WithDashboardResources = (WrappedComponent: React.ComponentType<DashboardItemProps>) =>
-  class WithDashboardResources_ extends React.Component<WithDashboardResourcesProps, WithDashboardResourcesState> {
+export const withDashboardResources = <P extends DashboardItemProps>(WrappedComponent: React.ComponentType<P>, additionalProps: any = {}) => (
+  connect(mapStateToProps, mapDispatchToProps)(class WithDashboardResources_ extends React.Component<WithDashboardResourcesProps, WithDashboardResourcesState> {
```
Comment (vojtechszocs) [inline frontend/public/components/dashboards-page/with-dashboard-resources.tsx]: Nit: for better readability, please make the `class WithDashboardResources_` declaration start on a new line.

**Comment by vojtechszocs** [comment]
LGTM with minor comment.

File: frontend/public/components/dashboards-page/with-dashboard-resources.tsx (hunk 2 above)
Comment (rawagner) [reply frontend/public/components/dashboards-page/with-dashboard-resources.tsx]: moved to new line. I also removed  `_` from `class WithDashboardResources_` as its not needed

File: frontend/public/components/dashboards-page/overview-dashboard/details-card.tsx [hunk 3]
```diff
@@ -33,16 +35,19 @@ const infrastructureResource: FirehoseResource = {
   prop: 'infrastructure',
 };
 
-export const DetailsCard_: React.FC<DetailsCardProps> = ({
+const mapStateToProps = (state: RootState) => ({
```
Comment (spadgett) [inline frontend/public/components/dashboards-page/overview-dashboard/details-card.tsx]: Prefer `connectToFlags` to keep the logic in one place

**Pull request merged**